pftp/proxy.go

```diff
@@ -185,23 +190,26 @@ func (s *proxyServer) sendTLSCommand(tlsProtocol uint16, previousTLSCommands []s
 		}
 
 		// read response
-		if _, err := s.originReader.ReadString('\n'); err != nil {
+		if _, err := reader.ReadString('\n'); err != nil {
 			return err
 		}
 
 		// SSL/TLS wrapping on connection
 		if strings.Contains(cmd, "AUTH") {
 			s.origin = tls.Client(s.origin, &config)
-			s.originReader = bufio.NewReader(s.origin)
-			s.originWriter = bufio.NewWriter(s.origin)
+			reader = bufio.NewReader(s.origin)
+			writer = bufio.NewWriter(s.origin)
 		}
 	}
 
+	*s.originReader = *reader
```

Comment:
why?
You should use a pointer.
```
s.originReadr = reader
```

Reply:
Oh, this was my fault.
now fixed at https://github.com/pyama86/pftp/pull/26/commits/4877eddc5b9eb0233abfbc8d79db440a57d9b92d